common/src/primitives/ip_address.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+const IPV4_SIZE: usize = 4;
+const IPV6_SIZE: usize = 16;
+
+/// IPv4 address.
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct Ip4 {
+    pub inner: std::net::Ipv4Addr,
+}
+
+/// IPv6 address.
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct Ip6 {
+    pub inner: std::net::Ipv6Addr,
```

Comment:
This doesn't need to be public anymore. Now you can use From/Into. In addition, you can implement AsRef to access the internal values.

Reply:
I replaced with fixed array and removed manual encode/decode implementation. Public removed. I think AsRef is not very useful now so not added.